temporalio/worker.py

```diff
@@ -1104,10 +1126,21 @@ def _heartbeat_processor(self) -> None:
                     fn(*item[1])
             except queue.Empty:
                 pass
-            except Exception as err:
+            except Exception:
                 logger.exception("Failed during multiprocess queue poll for heartbeat")
                 return
 
+    def flush_pending_heartbeats(self, task_token: bytes) -> None:
+        # We're gonna go ahead and just flush them all
```

Comment:
This is admittedly lazy and can result in issues if the queue fills up faster than it can be processed. The processing should be quick since it's just scheduling heartbeats with asyncio, not even converting them or anything.

For the next phase when workflow workers come about, I will probably move all the activity stuff to its own private file/package. I may also provide an alternative `SharedStateManager` that is a local loopback gRPC server instead of using `multiprocessing.Manager()` to see if that works better.